src/include/OpenImageIO/simd.h

```diff
@@ -2469,6 +2480,132 @@ OIIO_FORCEINLINE float4 nmsub (const simd::float4& a, const simd::float4& b,
+// Full precision log() of all components of a SIMD vector.
+OIIO_FORCEINLINE float4 log (const float4& v)
+{
+#if OIIO_SIMD_SSE
+    // Implementation inspired by:
+    // https://github.com/embree/embree/blob/master/common/simd/sse_special.h
+    // Which is listed as Copyright (C) 2007  Julien Pommier and distributed
+    // under the zlib license.
+    float4 x = v;
+    int4 emm0;
+    float4 zero (float4::Zero());
+    float4 one (1.0f);
+    mask4 invalid_mask = (x <= zero);
+    OIIO_SIMD_INT4_CONST (min_norm_pos, (int)0x00800000);
+    OIIO_SIMD_INT4_CONST (inv_mant_mask, (int)~0x7f800000);
+    x = max(x, bitcast_to_float4(int4(min_norm_pos)));  /* cut off denormalized stuff */
+    emm0 = srl (bitcast_to_int4(x), 23);
+    /* keep only the fractional part */
+    x = bitcast_to_float4 (bitcast_to_int4(x) & int4(inv_mant_mask));
+    x = bitcast_to_float4 (bitcast_to_int4(x) | bitcast_to_int4(float4(0.5f)));
+    emm0 = emm0 - int4(0x7f);
+    float4 e (emm0);
+    e = e + one;
+    OIIO_SIMD_FLOAT4_CONST (cephes_SQRTHF, 0.707106781186547524f);
+    mask4 mask = (x < float4(cephes_SQRTHF));
+    float4 tmp = bitcast_to_float4 (bitcast_to_int4(x) & bitcast_to_int4(mask));
+    x = x - one;
+    e = e - bitcast_to_float4 (bitcast_to_int4(one) & bitcast_to_int4(mask));
+    x = x + tmp;
+    float4 z = x * x;
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p0, 7.0376836292E-2f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p1, - 1.1514610310E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p2, 1.1676998740E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p3, - 1.2420140846E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p4, + 1.4249322787E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p5, - 1.6668057665E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p6, + 2.0000714765E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p7, - 2.4999993993E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_p8, + 3.3333331174E-1f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_q1, -2.12194440e-4f);
+    OIIO_SIMD_FLOAT4_CONST (cephes_log_q2, 0.693359375f);
+    float4 y = *(float4*)cephes_log_p0;
+    y = madd (y, x, float4(cephes_log_p1));
+    y = madd (y, x, float4(cephes_log_p2));
+    y = madd (y, x, float4(cephes_log_p3));
+    y = madd (y, x, float4(cephes_log_p4));
+    y = madd (y, x, float4(cephes_log_p5));
+    y = madd (y, x, float4(cephes_log_p6));
+    y = madd (y, x, float4(cephes_log_p7));
+    y = madd (y, x, float4(cephes_log_p8));
+    y = y * x;
+    y = y * z;
+    y = madd(e, float4(cephes_log_q1), y);
+    y = nmadd (z, 0.5f, y);
+    x = x + y;
+    x = madd (e, float4(cephes_log_q2), x);
+    x = bitcast_to_float4 (bitcast_to_int4(x) | bitcast_to_int4(invalid_mask)); // negative arg will be NAN
+    return x;
+#else
+    return float4 (std::log(a[0]), std::log(a[1]), std::log(a[2]), std::log(a[3]));
+#endif
+}
+
+
+
+// Full-precision pow(x,y) of all lanes of a SIMD vector.
+OIIO_FORCEINLINE float4 pow (const float4& x, const float4& y) {
+    return exp(log(abs(x)) * y);
```

Comment:
That might be an oversight. I see that for fast_pow (in fmath.h), I multiplied by sign(x). I think I meant to do that for the full-precision pow as well.